server/oslc-ui-model/src/main/java/org/eclipse/lyo/server/ui/model/PreviewFactory.java

```diff
@@ -68,12 +68,25 @@ public static Preview getPreview(final AbstractResource aResource, List<String>
                     value = constructPropertyValue(PropertyDefintion.RepresentationType.LINK, multiple, l);
                 } else {
                     Link link = (Link) getterMethod.invoke(aResource);
-                    value = constructPropertyValue(PropertyDefintion.RepresentationType.LINK, multiple,
-                        constructLink(link));
+                    value = constructPropertyValue(PropertyDefintion.RepresentationType.LINK, multiple, constructLink(link));
                 }
             } else {
-                value = constructPropertyValue(PropertyDefintion.RepresentationType.TEXT, multiple,
-                    getterMethod.invoke(aResource));
+            	if (!isNotInlinedRepresentation) {
+                    if (multiple) {
+                        Collection<AbstractResource> rs = (Collection<AbstractResource>) getterMethod.invoke(aResource);
```

Comment:
do we expect both the invoke and the cast to never throw an exception?

Reply:
are you suggesting I fix it?

Reply:
The logic is based on the OSLC annotations and hence indirectly assuming we have a collection for example.
Maybe better then to wrap with exception handling.
I am tempted however to wrap this whole thing with 1 exception handler.

Reply:
Hi @berezovskyi ! I look at this again. We have similar calls in the rest of the method.
We don't catch any exceptions. But the function throws IllegalAccessException, IllegalArgumentException, InvocationTargetException, NoSuchMethodException, SecurityException 

So inline we that, we don't need to fix this. 
Unless of course, we fix it in the rest of the method.

Reply:
Sounds good. NoSuchMethodException, SecurityException are obvious ones, don't think we need to handle them. IllegalAccessException, IllegalArgumentException may be worth wrapping into Lyo exceptions in the future.